src/examples/algorithms/HelloSP_TP.cpp

```diff
@@ -0,0 +1,87 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013-2015, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+#include <iostream>
+#include <vector>
+#include <algorithm>    // std::generate
+#include <ctime>        // std::time
+#include <cstdlib>      // std::rand, std::srand
+#include <cmath> 	// pow
+
+#include "nupic/algorithms/SpatialPooler.hpp"
+#include "nupic/algorithms/Cells4.hpp"
+
+using namespace std;
+using namespace nupic;
+using namespace nupic::algorithms::spatial_pooler;
```

Comment:
For SP and Cells4 can you bring the specific class into scope rather than `using namespace`? Like this: `using nupic::algorithms::spatial_pooler::SpatialPooler;` to avoid bringing something else into scope accidentally.